build_wheel.sh

```diff
@@ -449,6 +449,34 @@ function adjust_numpy_requirements_based_on_link_info()
 		fi
 	fi
 }
+function adjust_torch_requirements_based_on_link_info()
+{
+	# don't modify torch wheels themselves
+	if [[ $WHEEL_NAME =~ torch-.* ]]; then
+		return
+	fi
+	# only linux_x86_64 wheels will contain .so'
+	if [[ $WHEEL_NAME =~ .*linux_x86_64.* ]]; then
+		tmpdir=/tmp/wheel_builder_$BASHPID_$RANDOM
+		echo "Testing if wheel depends on torch..."
+		$SCRIPT_DIR/manipulate_wheels.py --print_req --wheels $TMP_WHEELHOUSE/$WHEEL_NAME | grep "^torch" >/dev/null
+		res=$?
+		if [[ $res -eq 0 ]]; then
+			echo "Dependency on torch found. Pinning version of torch"
+			torch_build_version=$(pip show torch | grep Version | awk '{print $2}' | sed -e "s/\([^+]*\)+*.*/\1/g")
+			torch_build_version_without_dots=${torch_build_version//./}
+			log_command $SCRIPT_DIR/manipulate_wheels.py --print_req --wheels $TMP_WHEELHOUSE/$WHEEL_NAME
+			tag="torch$torch_build_version_without_dots"
+			log_command $SCRIPT_DIR/manipulate_wheels.py --inplace --force --wheels $TMP_WHEELHOUSE/$WHEEL_NAME --update_req "\"torch (==$torch_build_version)\"" --add_tag $tag 
```

Comment:
Mmm, why ?

Reply:
The build script has no way of knowing if there are conflicting versions or not

Reply:
Because it is simply not necessary in most cases, only when we have two versions that would build with a different dependency. That's the reason why we added the tag in the first place. torch-scatter is an example of this. And that's up to a human to determine this based on the package version being build.

Other wheels already depends on the specific version from the requirements and thus do not need any tags to differentiate them.

Reply:
Might not be needed, but it is not harmful ? More thinking means more errors. More automation means less error.

Reply:
I agree. It is more to avoid appending non necessary information when not required.
Pip being pip, it has some issues with local tags sometimes.